system-addon/lib/PlacesFeed.jsm

```diff
@@ -100,13 +100,19 @@ class BookmarksObserver extends Observer {
    * @param  {str} title
    * @param  {int} dateAdded
    * @param  {str} guid      The unique id of the bookmark
+   * @param  {str} parent guid
+   * @param  {int} source    Used to distinguish bookmarks made by different
+   *                         actions: sync, bookmarks import, other.
    */
   onItemAdded(...args) {
     const type = args[3];
-    if (type !== PlacesUtils.bookmarks.TYPE_BOOKMARK) {
+    const source = args[9];
+    const uri = args[4];
+    if (type !== PlacesUtils.bookmarks.TYPE_BOOKMARK ||
+        source !== PlacesUtils.bookmarks.SOURCES.DEFAULT ||
```

Comment:
We should have comments on why we're doing these checks. And we probably should skip for `SOURCES.IMPORT_REPLACE` instead of skip if not `SOURCES.DEFAULT` as we haven't run into bugs for other types of sources but filtering out too much might introduce new bugs.

Reply:
👍